src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -11,6 +11,7 @@
 # distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
+import inspect
```

Comment:
This here is being imported but never used, did you run pre-commit on this branch?